processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -241,7 +241,11 @@ public ExpressionType getOutputType(Expr.InputBindingInspector inspector, List<E
     @Override
     public boolean canVectorize(Expr.InputBindingInspector inspector, List<Expr> args)
     {
-      return inspector.areNumeric(args) && inspector.canVectorize(args);
+      final ExpressionType outputType = args.get(0).getOutputType(inspector);
+      if (outputType == null && NullHandling.replaceWithDefault()) {
+        return false;
```

Comment:
worth leaving a comment about why this is the case. something about how non-vectorized treats null values as `STRING` types, which makes it kind of complicated to achieve consistent behavior for vector and non-vector engines

Reply:
added